### PR TITLE
NAS-117236 / 22.12-RC.1 / Check for keyboard support on new storage pages

### PR DESCRIPTION
**Pull request by AlexKarpov98**
checked accessibility via keyboard, made sure everything highlights and responds to user keyboard actions.

Screen Shot 2022-10-20 at 13 01 00

### CODE REVIEW

**Review by undsoft — CHANGES_REQUESTED** [review]
Cannot select device in Devices, cannot select a dataset in Datasets.

**Comment by AlexKarpov98** [comment]
Fixed hovering effect for **tree-item**
Screen Shot 2022-10-21 at 15 19 12

And added **enter press** support to view tree-item's data.

File: src/app/pages/datasets/components/dataset-management/dataset-management.component.html [hunk 1]
```diff
@@ -86,6 +87,7 @@
                 routerLinkActive="selected"
                 [routerLink]="['/datasets', dataset.id]"
                 (click)="openMobileDetails()"
+                (keydown.enter)="openMobileDetails(['/datasets', dataset.id])"
```
Comment (undsoft) [inline src/app/pages/datasets/components/dataset-management/dataset-management.component.html:90]: It looks like you can unify all of this: 
1. Remove routerLink directive.
2. Have the same handler for (click) and (keydown.enter).
3. Call it something else like `onRowSelected(dataset: Dataset)`

**Review by undsoft — APPROVED** [review]
If I am in a smaller screen and have mobile layout, it seems like focus stays in the table and it's hard to start using elements in the mobile view (I have to Tab through the whole table without seeing anything happening).
See if there is an easy fix for this. If there is no simple fix, you can merge without it.

**Comment by AlexKarpov98** [comment]
@undsoft - have a look at it now please :)

**Review by undsoft — APPROVED** [review]
There is a button to expand/collapse rows. It is no longer possible to expand row on mobile without going to mobile details.
That is if you have a collapsed row and you want to select a child, it's not going to be easy.

Otherwise works okay.

**Comment by AlexKarpov98** [comment]
tested and now works correctly

**Review by undsoft — APPROVED** [review]
Works okay.
I would do
```
(keydown.enter)="$event.stopPropagation()"
```
on button instead of looking at element tag. Less fragile.

**Pull request merged**